### PR TITLE
Add drinking hat

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

```SUMMARY: Content "Add drinking hat"```
#### Purpose of change

Add some additional way to make use of head slot to wear something useful.
#### Describe the solution

Add drinking hat and recipe for it.
#### Describe alternatives you've considered

None.
#### Testing

Spawned drinking hat, verified that player can't store drinking hat in their inventory as long as it has liquid in it, and can wear it if it does have liquid inside.
#### Additional context

None.

### CODE REVIEW

**Comment by Tsunder** [comment]
context: #29191 Drinking hats

> If this gets closed, I think I'll recycle this to make just one drinking hat item.

**Comment by RickyVanz** [comment]
i can hear spongebob saying "he is number one" from far away.. would love to try this in-game.

File: data/json/recipes/armor/head.json [hunk 1]
```diff
@@ -120,6 +120,32 @@
     "tools": [ [ [ "tongs", -1 ] ], [ [ "swage", -1 ] ], [ [ "forge", 50 ], [ "oxy_torch", 10 ] ] ],
     "components": [ [ [ "gold_small", 300 ] ] ]
   },
+  {
+    "type": "recipe",
+    "result": "drinking_hat",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_HEAD",
+    "skill_used": "fabrication",
```
Comment (snipercup) [inline data/json/recipes/armor/head.json]:
```suggestion
    "skill_used": "fabrication",
    "difficulty": 1,
```

Should be fab 1

File: data/json/recipes/armor/head.json [hunk 2]
```diff
@@ -120,6 +120,32 @@
...
+        [ "can_food_unsealed", 2 ],
+        [ "canister_empty", 2 ],
+        [ "bottle_plastic_small", 2 ],
+        [ "sippy_cup", 2 ],
+        [ "clay_canister", 2 ],
+        [ "flask_glass", 2 ],
+        [ "beaker", 2 ],
+        [ "flask_hip", 2 ]
+      ],
+      [ [ "hose", 1 ] ]
```
Comment (snipercup) [inline data/json/recipes/armor/head.json]:
```suggestion
      [ [ "hose", 1 ] ],
      [ [ "wire", 2 ] ]
```

I imagine it's some kind of headband with cans strapped to a ring? Wires should help with that.

Comment (Tsunder) [reply data/json/recipes/armor/head.json]: previously, the binding material was duct tape :)

File: data/json/recipes/armor/head.json [hunk 3]
```diff
@@ -120,6 +120,32 @@
...
+      [
+        [ "can_drink_unsealed", 2 ],
+        [ "can_food_unsealed", 2 ],
+        [ "canister_empty", 2 ],
+        [ "bottle_plastic_small", 2 ],
+        [ "sippy_cup", 2 ],
+        [ "clay_canister", 2 ],
+        [ "flask_glass", 2 ],
+        [ "beaker", 2 ],
+        [ "flask_hip", 2 ]
```
Comment (Tsunder) [inline data/json/recipes/armor/head.json]: could split it into two ingredient groups of both requiring 1 bottle-like, to allow mixing.

File: data/json/recipes/armor/head.json (hunk 2 above)
Comment (Rail-Runner) [reply data/json/recipes/armor/head.json]: I went with 1 unit of wire; judging by weight, 2 units seem to be more than would be needed.

**Comment by Rail-Runner** [comment]
Well, turns out one can't use same requirement twice in a recipe. Had to work around that and do the same without using requirements.

**Pull request merged**